src/kakarot/execution_context.cairo

```diff
@@ -356,16 +394,25 @@ namespace ExecutionContext {
     // @dev Extract the byte that the current pc is pointing to and revert if it is not a JUMPDEST operation.
     // @param self The pointer to the execution context
     // @param pc_location location to check
-    func check_jumpdest(self: model.ExecutionContext*, pc_location: felt) {
+    func check_jumpdest{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(self: model.ExecutionContext*, pc_location: felt) {
         alloc_locals;
-        let (local output: felt*) = alloc();
+        // let (local output: felt*) = alloc();
 
         // Copy bytecode slice
-        memcpy(dst=output, src=self.call_context.bytecode + pc_location, len=1);
+        let (res, rem) = unsigned_div_rem(pc_location, 31);
+        let value: felt = [self.call_context.bytecode + res];
+        let output: felt = get_byte_in_array(
+            offset=rem, felt_packed_code=value, return_byte_length=1
+        );
```

Comment:
isn't very costly in terms of steps when executing a bytecode? I'd like to see the final updated benchmark (resources.csv file generated by the test suite) to be convinced!!

Reply:
So it is a bit costly but the idea that it would be a tradeoff between storing 31 bytes in different slots. But as of now the whole process is less efficient after the last changes, so I need to change a few things